internal/server/v3/node/golden/node_test.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package golden
+
+import (
+	"context"
+	"path"
+	"runtime"
+	"testing"
+
+	pbs "github.com/datacommonsorg/mixer/internal/proto/service"
+	pbv3 "github.com/datacommonsorg/mixer/internal/proto/v3"
+	"github.com/datacommonsorg/mixer/test"
+	"github.com/google/go-cmp/cmp"
+	"google.golang.org/protobuf/testing/protocmp"
+)
+
+func TestV3Node(t *testing.T) {
+	// TODO: Remove check once enabled.
+	if !test.EnableSpannerGraph {
+		return
+	}
+	t.Parallel()
+	ctx := context.Background()
+
+	_, filename, _, _ := runtime.Caller(0)
+	goldenPath := path.Dir(filename)
+
+	testSuite := func(mixer pbs.MixerClient, latencyTest bool) {
```

Comment:
nit: Consider using just datasources vs the full mixer client for tests at this level.

Reply:
I added this just to test the full e2e for mixer v3, but I'll add all other node tests on the datasource level (I'll put them in datasources + client specific datasource)